### PR TITLE
ppl: update annealing to support pin constraints

### PR DESCRIPTION
**Pull request by eder-matheus**

### CODE REVIEW

File: src/ppl/src/IOPlacer.cpp [hunk 1]
```diff
@@ -1507,11 +1507,12 @@ void IOPlacer::addVerLayer(odb::dbTechLayer* layer)
...
 }
 
-void IOPlacer::getPinsFromDirectionConstraint(Constraint& constraint)
+void IOPlacer::getPinsFromDirectionConstraint(Constraint& constraint,
+                                              int constraint_idx)
 {
   if (constraint.direction != Direction::invalid
       && constraint.pin_list.empty()) {
-    for (const IOPin& io_pin : netlist_io_pins_->getIOPins()) {
+    for (IOPin& io_pin : netlist_io_pins_->getIOPins()) {
```
Comment (maliberty) [inline src/ppl/src/IOPlacer.cpp:1515]: why drop the const?

File: src/ppl/src/IOPlacer.cpp [hunk 2]
```diff
@@ -1507,11 +1507,12 @@ void IOPlacer::addVerLayer(odb::dbTechLayer* layer)
   ver_layers_.insert(layer->getRoutingLevel());
 }
 
-void IOPlacer::getPinsFromDirectionConstraint(Constraint& constraint)
+void IOPlacer::getPinsFromDirectionConstraint(Constraint& constraint,
+                                              int constraint_idx)
```
Comment (maliberty) [inline src/ppl/src/IOPlacer.cpp:1511]: How is the new argument used?

**Review by maliberty — COMMENTED** [review]
Is this handling mirroring or only grouping?

File: src/ppl/src/IOPlacer.cpp (hunk 2 above)
Comment (eder-matheus) [reply src/ppl/src/IOPlacer.cpp]: Fixed this and the above/ I've started with these two changes, but they're not necessary anymore.

**Comment by eder-matheus** [comment]
> Is this handling mirroring or only grouping?

For this PR, only region constraints for lone pins and groups. Mirroring is my next (and probably last) step to have a complete version of PPL with Simulated Annealing.

**Review by maliberty — APPROVED** [review]

**Pull request merged**